kv/dist_sender.go

```diff
@@ -684,30 +689,29 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
 		if finalReply != curReply {
 			// This was the second or later call in a multi-range request.
 			// Combine the new response with the existing one.
-			_, ok := finalReply.(proto.Combinable)
-			if !ok {
+			if cFinalReply, ok := finalReply.(proto.Combinable); ok {
+				cFinalReply.Combine(curReply)
+			} else {
 				// This should never apply in practice, as we'll only end up here
 				// for range-spanning requests.
-				call.Reply.Header().SetGoError(
-					util.Errorf("multi-range request with non-combinable response type"))
+				call.Reply.Header().SetGoError(util.Errorf("multi-range request with non-combinable response type"))
 				return
 			}
-			finalReply.(proto.Combinable).Combine(curReply)
 		}
 
 		// If this request has a bound, such as MaxResults in
 		// ScanRequest, check whether enough rows have been retrieved.
-		var prevBound int64
-		if args, ok := args.(proto.Bounded); ok && args.GetBound() > 0 {
-			prevBound = args.GetBound()
-			if cReply, ok := curReply.(proto.Countable); ok {
-				if nextBound := prevBound - cReply.Count(); nextBound > 0 {
-					// Update bound for the next round.
-					// We've deferred restoring the original bound earlier.
-					args.SetBound(nextBound)
-				} else {
-					// Set flag to break the loop.
-					descNext = nil
+		if boundedArgs != nil {
```

Comment:
I'm not sure if it's an improvement to create a new boundedArgs variable whose lifetime spans 60+ lines of code. I think I'd rather leave this part as it was.